File: java-spring/hexagonal-feed/hexagonal-feed-infrastructure/src/main/java/com/example/hexagonalfeed/common/error/exception/CustomException.java
package com.example.hexagonalfeed.common.error.exception;

import com.example.hexagonalfeed.common.error.ErrorCode;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class CustomException extends RuntimeException {
    private final ErrorCode errorCode;
}

File: kotlin-spring/feed/src/main/kotlin/com/example/feed/global/error/exception/CustomException.kt
package com.example.feed.global.error.exception

import com.example.feed.global.error.response.ErrorResponse

abstract class CustomException(
        private val error: ErrorResponse
): RuntimeException() {

    val status: Int
        get() = error.status

    override val message: String
        get() = error.message
}